src/scripts/main.js

```diff
@@ -1,3 +1,71 @@
 'use strict';
 
 // write code here
+document.addEventListener('DOMContentLoaded', () => {
+  const table = document.querySelector('table');
+  const appendRowBtn = document.querySelector('.append-row');
+  const removeRowBtn = document.querySelector('.remove-row');
+  const appendColumnBtn = document.querySelector('.append-column');
+  const removeColumnBtn = document.querySelector('.remove-column');
+
+  const maxRows = 10;
+  const minRows = 2;
+  const maxColumns = 10;
+  const minColumns = 2;
+
+  function updateButtons() {
+    const rowCount = table.rows.length;
+    const colCount = table.rows[0].cells.length;
+
+    appendRowBtn.disabled = rowCount >= maxRows;
+    removeRowBtn.disabled = rowCount <= minRows;
+    appendColumnBtn.disabled = colCount >= maxColumns;
+    removeColumnBtn.disabled = colCount <= minColumns;
+  }
+
+  appendRowBtn.addEventListener('click', () => {
+    const rowCount = table.rows.length;
+
+    if (rowCount < maxRows) {
+      const newRow = table.insertRow();
+
+      for (let i = 0; i < table.rows[0].cells.length; i++) {
+        newRow.insertCell();
+      }
+      updateButtons();
+    }
+  });
+
+  removeRowBtn.addEventListener('click', () => {
+    const rowCount = table.rows.length;
+
+    if (rowCount > minRows) {
+      table.deleteRow(-1);
+      updateButtons();
+    }
+  });
+
+  appendColumnBtn.addEventListener('click', () => {
+    const colCount = table.rows[0].cells.length;
```

Comment:
Similar to the previous comment, ensure that the table has at least one row before accessing `table.rows[0].cells.length` to avoid errors.